pulp_2_tests/tests/docker/api_v2/test_remove.py

```diff
@@ -21,123 +197,342 @@ class RemoveV2ContentTestCase(unittest.TestCase):
+
+        # Delete by SHA256 in batch
+        self.delete_docker_units(repo, DOCKER_V2_MANIFEST)
 
         # Count the remaining units
-        remaining_units = self.get_docker_units(self.repo, 'docker_manifest')
-        self.assertEqual(len(remaining_units), 0, remaining_units)
+        self.count_docker_units(repo, 'NONE')
+
+    def test_06_remove_blob_all_batch(self):
+        """Sync docker repo and remove all blobs in batch."""
+        # Batch requires 2.21 hot-fix patch in #4549
+        if self.cfg.pulp_version < Version('2.21'):
+            raise unittest.SkipTest(
+                'This test requires Pulp 2.21 or newer.'
+            )
+
+        # Create and Verify initial unit count
+        repo = self.create_and_copy_test_repo(self.repo, DOCKER_V2_BLOB)
+        self.addCleanup(self.client.delete, repo['_href'])
+        self.count_docker_units(repo, 'ALL_V2_BLOB_COPY')
+
+        # Delete by SHA256 in batch
+        self.delete_docker_units(repo, DOCKER_V2_BLOB)
 
-    # pylint: disable=R0201
-    def test_04_remove_manifest_list_not_shared(self):
+        # Count the remaining units
+        self.count_docker_units(repo, 'NONE')
+
+    def test_07_remove_manifest_list_not_shared(self):
         """Sync docker repo and remove some non-shared manifest_lists."""
-        raise unittest.SkipTest('Stubbed test case, Not Implemented Yet')
+        # Create and Verify initial unit count
+        repo = self.create_and_copy_test_repo(
+            self.repo, NON_SHARED_MANIFEST_LIST
+        )
+        self.addCleanup(self.client.delete, repo['_href'])
+        self.count_docker_units(repo, 'NON_SHARED_MANIFEST_LIST')
 
-    def test_05_remove_manifest_not_shared(self):
-        """Sync docker repo and remove some non-shared manifests."""
-        raise unittest.SkipTest('Stubbed test case, Not Implemented Yet')
+        # Delete by SHA256 in batch of 1 SHA256
+        single_non_shared_manifest_list = []
+        single_non_shared_manifest_list.append(NON_SHARED_MANIFEST_LIST[0])
+        self.delete_docker_units(repo, single_non_shared_manifest_list)
 
-    def test_06_remove_manifest_list_shared(self):
+        # Count the remaining unshared units from the second manifest.
+        self.count_docker_units(repo, 'NON_SHARED_MANIFEST_LIST_DELETE')
+
+    def test_08_remove_manifest_list_shared(self):
         """Sync docker repo and remove some shared manifest_lists."""
-        raise unittest.SkipTest('Stubbed test case, Not Implemented Yet')
+        # Create and Verify initial unit count
+        repo = self.create_and_copy_test_repo(self.repo, SHARED_MANIFEST_LIST)
+        self.addCleanup(self.client.delete, repo['_href'])
+        self.count_docker_units(repo, 'SHARED_MANIFEST_LIST')
+
+        # Delete by SHA256 in batch of 1 SHA256
+        single_shared_manifest_list = []
+        single_shared_manifest_list.append(SHARED_MANIFEST_LIST[0])
+        self.delete_docker_units(repo, single_shared_manifest_list)
```

Comment:
If the only reason why we are creating a new list is to cast...perhaps we could make a bit simpler.
```suggestion
        self.delete_docker_units(repo, [SHARED_MANIFEST_LIST[0]])
```

And if the order of the SHARED_MANIFEST_LIST does not matter.

Assuming that choice was imported, we could even do. 

```suggestion
        self.delete_docker_units(repo, [choice(SHARED_MANIFEST_LIST)])
```

I tested those changes locally.

Reply:
TIL python casting.